src/BaGet.Core/PackageDatabase.cs

```diff
@@ -11,24 +11,28 @@ namespace BaGet.Core
     public class PackageDatabase : IPackageDatabase
     {
         private readonly IContext _context;
+        private readonly Func<IContext> _newContext;
 
-        public PackageDatabase(IContext context)
+        public PackageDatabase(IContext context, Func<IContext> newContext)
         {
             _context = context ?? throw new ArgumentNullException(nameof(context));
+            _newContext = newContext ?? throw new ArgumentNullException(nameof(newContext));
         }
 
         public async Task<PackageAddResult> AddAsync(Package package, CancellationToken cancellationToken)
         {
+            using var context = _newContext();
```

Comment:
This read as confusing with the field _context and this.

Why is the field still required? Could it not always generate context?

Reply:
Agree and have updated this. Would like to change this everywhere to be honest. I don't like relying on DI to manage DbContext lifetime.